core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -777,6 +808,35 @@ private[spark] object SparkConf extends Logging {
     }
   }
 
+  /**
+   * A function to help parsing configs with multiple parts where the base and
+   * suffix could be one of many options. For instance configs like:
+   * spark.executor.resource.{resourceName}.{count/addresses}
+   * This function takes an Array of configs you got from the
+   * getAllWithPrefix function, selects only those that end with the suffix
+   * passed in and returns just the base part of the config before the first
+   * '.' and its value.
+   */
+  def getConfigsWithSuffix(
```

Comment:
Ditto. This is also too specific. Looking at the method name,  I cannot guess what the method does, in particular, it also change the key in the return result.

Reply:
ok what do you suggest?

Reply:
I can just move them into the Resourcediscoverer object and try to rename or make a ResourceConfUtils class, would like your input before changing so we don't go back and forth a bunch of times